src/Altinn.AccessManagement.Persistence/DelegationMetadataRepo.cs

```diff
@@ -0,0 +1,779 @@
+        {
+            if (resourceMatchType == ResourceAttributeMatchType.AltinnAppId)
+            {
+                return await InsertAppDelegation(delegationChange, cancellationToken);
+            }
+
+            return await InsertResourceRegistryDelegation(delegationChange, cancellationToken);
+        }
+
+        private async Task<DelegationChange> InsertAppDelegation(DelegationChange delegationChange, CancellationToken cancellationToken = default)
+        {
+            using var activity = TelemetryConfig.ActivitySource.StartActivity(ActivityKind.Client);
+
+            var param = new Dictionary<string, object>
+            {
+                { "delegationChangeType", delegationChange.DelegationChangeType },
+                { "altinnAppId", delegationChange.ResourceId },
+                { "offeredByPartyId", delegationChange.OfferedByPartyId },
+                { "coveredByUserId", delegationChange.CoveredByUserId.HasValue ? delegationChange.CoveredByUserId.Value : DBNull.Value },
+                { "coveredByPartyId", delegationChange.CoveredByPartyId.HasValue ? delegationChange.CoveredByPartyId.Value : DBNull.Value },
+                { "performedByUserId", delegationChange.PerformedByUserId.HasValue ? delegationChange.PerformedByUserId.Value : DBNull.Value },
+                { "blobStoragePolicyPath", delegationChange.BlobStoragePolicyPath },
+                { "blobStorageVersionId", delegationChange.BlobStorageVersionId }
+            };
+
+            string query = $"INSERT INTO delegation.delegationChanges(delegationChangeType, altinnAppId, offeredByPartyId, coveredByUserId, coveredByPartyId, performedByUserId, blobStoragePolicyPath, blobStorageVersionId)" +
+                $"VALUES (@delegationChangeType, @altinnAppId, @offeredByPartyId, @coveredByUserId, @coveredByPartyId, @performedByUserId, @blobStoragePolicyPath, @blobStorageVersionId)" +
+                $"RETURNING *;";
+
+            try
+            {
+                var res = await _connection.QueryAsync<DelegationChange>(new CommandDefinition(query, param, cancellationToken: cancellationToken));
+                return res.FirstOrDefault();
+            }
+            catch (Exception ex)
+            {
+                activity?.StopWithError(ex);
+                throw;
+            }
+            finally
+            {
+                _connection.Close();
+            }
+        }
+
+        private async Task<DelegationChange> InsertResourceRegistryDelegation(DelegationChange delegationChange, CancellationToken cancellationToken = default)
+        {
+            using var activity = TelemetryConfig.ActivitySource.StartActivity(ActivityKind.Client);
+            var param = new Dictionary<string, object>
+            {
+                { "delegationChangeType", delegationChange.DelegationChangeType },
+                { "altinnAppId", delegationChange.ResourceId }, // resourceregistryid ?
```

Comment:
yes, `delegationChange.ResourceId` should be correct based on existing implementation.

Reply:
Ok